File: app/src/main/java/com/comp318_002_g5/voicetodo/ToDoList.java
package com.comp318_002_g5.voicetodo;

import java.util.ArrayList;


public class ToDoList {
    public String Name;
    public ArrayList<ToDoListItem> Tasks;
    public ToDoList(String name) {
        Name = name;
        Tasks = new ArrayList<>();
    }
    public void AddTask(String task){
        Tasks.add(new ToDoListItem(task));
    }

    public void AddTasks(ArrayList<String> tasks){
        for(String task: tasks){
            AddTask(task);
        }
    }
    public void ToggleComplete(int idx){
        // Setting this up with Index reference if the UI is easier
        // to interact with as a list of elements
        Tasks.get(idx).ToggleComplete();
    }

    public String PrintTasks(){
        StringBuilder out_msg = new StringBuilder();
        for(ToDoListItem task: Tasks){
            out_msg.append("- ").append(task.Task).append("/n");
        }
        return out_msg.toString();
    }
}

File: lib/classes/ToDoList.py
class ToDoList:
    tasks = []

    class ToDoTask:
        def __init__(self, task):
            self.desc = task
            self.completed = False

        def toggle_complete(self):
            self.completed = not self.completed

    def __init__(self):
        self.tasks = []

    def add_task(self, task_desc):
        self.tasks.append(self.ToDoTask(task_desc))

    def rem_task(self, idx):
        self.tasks.pop(idx)

    def __str__(self):
        out_msg = ""
        for task in self.tasks:
            x_or_o = "[X]" if task.completed else "[ ]"
            out_msg += f"- {x_or_o} {task.desc} \n"
        return out_msg
